src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -447,6 +462,81 @@ def _build_extended_key_usage(self, ext):
 
         return x509.ExtendedKeyUsage(ekus)
 
+    def _build_crl_distribution_points(self, ext):
+        cdps = self._backend._ffi.cast(
+            "Cryptography_STACK_OF_DIST_POINT *",
+            self._backend._lib.X509V3_EXT_d2i(ext)
+        )
+        assert cdps != self._backend._ffi.NULL
+        cdps = self._backend._ffi.gc(
+            cdps, self._backend._lib.sk_DIST_POINT_free)
+        num = self._backend._lib.sk_DIST_POINT_num(cdps)
+
+        dist_points = []
+        for i in range(num):
+            full_name = None
+            relative_name = None
+            crl_issuer = None
+            reasons = None
+            cdp = self._backend._lib.sk_DIST_POINT_value(cdps, i)
+            if cdp.reasons != self._backend._ffi.NULL:
+                reasons = []
+                # We will check each bit from RFC 5280
+                # ReasonFlags ::= BIT STRING {
+                #      unused                  (0),
+                #      keyCompromise           (1),
+                #      cACompromise            (2),
+                #      affiliationChanged      (3),
+                #      superseded              (4),
+                #      cessationOfOperation    (5),
+                #      certificateHold         (6),
+                #      privilegeWithdrawn      (7),
+                #      aACompromise            (8) }
+                for bit in range(1, 9):
+                    if self._backend._lib.ASN1_BIT_STRING_get_bit(
+                        cdp.reasons, bit
+                    ):
```

Comment:
Ugh, right. Let me think.

On Tue, May 12, 2015 at 11:49 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/hazmat/backends/openssl/x509.py
> <https://github.com/pyca/cryptography/pull/1920#discussion_r30200778>:
>
> > +                reasons = []
> > +                # We will check each bit from RFC 5280
> > +                # ReasonFlags ::= BIT STRING {
> > +                #      unused                  (0),
> > +                #      keyCompromise           (1),
> > +                #      cACompromise            (2),
> > +                #      affiliationChanged      (3),
> > +                #      superseded              (4),
> > +                #      cessationOfOperation    (5),
> > +                #      certificateHold         (6),
> > +                #      privilegeWithdrawn      (7),
> > +                #      aACompromise            (8) }
> > +                for bit in range(1, 9):
> > +                    if self._backend._lib.ASN1_BIT_STRING_get_bit(
> > +                        cdp.reasons, bit
> > +                    ):
>
> If we want a series of if statements I'll have to put a new vector in that
> sets every bit and also one that tests the negative cases not covered by
> the existing vectors. Obviously doable, but required if we want to maintain
> coverage.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1920/files#r30200778>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084